examples/aggregator/module/rust/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Aggregator example for Project Oak.
+//!
+//! This shows how an Oak Node can aggregate data samples and expose the aggregated data only if
+//! there are enough samples to hide individual contributors.
+//!
+//! Clients invoke the module by providing a vector of non-negative numbers and get back an
+//! aggregated vector if an Oak Node has collected more samples than the predefined threshold.
+
+mod aggregator;
+mod proto;
+#[cfg(test)]
+mod tests;
+
+use aggregator::{Monoid, ThresholdAggregator};
+use log::{info, warn};
+use oak::grpc;
+use proto::aggregator::Vector;
+use proto::aggregator_grpc::{dispatch, Aggregator};
+use protobuf::well_known_types::Empty;
+
+const SAMPLE_THRESHOLD: u64 = 3;
+
+oak::entrypoint!(oak_main => {
+    oak_log::init_default();
+    AggregatorNode {
+        aggregator: ThresholdAggregator::<Vector>::new(SAMPLE_THRESHOLD),
+    }
+});
+
+impl Monoid for Vector {
+    fn identity() -> Self {
+        Vector::new()
+    }
+
+    /// Adds two non-negative integer vectors elementwise. If vectors have different lengths, then
+    /// pads the smallest vector with zeros.
+    fn combine(&self, other: &Self) -> Self {
+        use itertools::EitherOrBoth::*;
+        use itertools::Itertools;
+
+        let mut vector = Vector::new();
+        vector.set_items(
```

Comment:
I'm not sure I understand why this is allocating a new vector, and collecting the iterator via `set_items`. Why not just return the result of `collect`?

Reply:
Code generated by `protobuf` doesn't implement this:
```
value of type `proto::aggregator::Vector` cannot be built from `std::iter::Iterator<Item=u64>`
```

Reply:
Done